packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -40,10 +40,11 @@ class UnknownMapObjectIdError extends Error {
 }
 
 /// Android specific settings for [GoogleMap].
-// TODO(stuartmorgan): Deprecate this in favor of pointing people who want to
-// change this to using the Android implementation Dart class directly. This
-// should be done as part of switching the default to hybrid composition.
+@Deprecated(
+    'See https://pub.dev/packages/google_maps_flutter_android#display-mode')
 class AndroidGoogleMapsFlutter {
+  @Deprecated(
```

Comment:
nit: Is it necessary to deprecate a private constructor.

Reply:
Yes, otherwise we get https://dart-lang.github.io/linter/lints/deprecated_consistency.html. At first I thought it was a bug, but then I realized that we're using `deprecated_member_use_from_same_package: ignore` in Flutter repos, and without that it would make sense because other code in the library could reference the private constructor.